controller/internal/cache/user_group.go

```diff
@@ -0,0 +1,51 @@
+package cache
+
+import (
+	cmap "github.com/orcaman/concurrent-map/v2"
+	"github.com/pluralsh/console/controller/internal/client"
+)
+
+type userGroupCache struct {
+	consoleClient client.ConsoleClient
+	userMap       cmap.ConcurrentMap[string, string]
+	groupMap      cmap.ConcurrentMap[string, string]
+}
+
+type UserGroupCache interface {
+	GetGroupID(name string) (string, error)
+	GetUserID(email string) (string, error)
+}
+
+func NewUserGroupCache(consoleClient client.ConsoleClient) UserGroupCache {
```

Comment:
we should also spawn a goroutine to periodically wipe this cache (maybe every 30m) to serve as a simple expiration mechanism

Reply:
done